### ISSUE TITLE
SC2154 second use of unassigned variable not reported

### ISSUE THREAD

**Issue by kvantour** [koalaman/shellcheck#1970]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154 
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
echo "$foo"
echo "$foo"

```

#### Here's what shellcheck currently says:
Line 2:
echo "$foo"
      ^-- SC2154: foo is referenced but not assigned.

#### Here's what I wanted or expected to see:
Line 2:
echo "$foo"
      ^-- SC2154: foo is referenced but not assigned.
Line 3:
echo "$foo"
      ^-- SC2154: foo is referenced but not assigned.

**Comment by koalaman** [comment]
Your assessment is obviously correct, but it's done this way intentionally for pragmatic reasons. The extra warning doesn't add much value in terms of helping someone fix the problem, and if ShellCheck was wrong and you want to disable the warning, this allows you to do so in a single place.
